CMakeLists.txt

```diff
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.26.4 FATAL_ERROR)
 
 project(
   pynvjitlink
-  VERSION 0.1.12
+  VERSION ${SKBUILD_PROJECT_VERSION}
```

Comment:
`scikit-build-core` [sets `${SKBUILD_PROJECT_VERSION}`]( https://scikit-build-core.readthedocs.io/en/latest/cmakelists.html#accessing-information ) for us

So just use that here